tests/xml.js

```diff
@@ -0,0 +1,128 @@
+require([
+    'jquery',
+    'chai',
+    'vellum/xml'
+], function (
+    $,
+    chai,
+    xml
+) {
+    var assert = chai.assert;
+
+    describe("The XML normalizer", function () {
+        function eq(value, escaped) {
+            assert.strictEqual(xml.normalize(value), escaped);
+        }
+
+        it("should preserve empty string", function () {
+            eq('', '');
+        });
+
+        it("should preserve space", function () {
+            eq(' ', ' ');
+        });
+
+        it("should preserve newline", function () {
+            eq(' ', ' ');
+        });
+
+        it("should preserve empty XML node", function () {
+            eq('<output />', '<output />');
+        });
+
+        it("should escape & character", function () {
+            eq('Mother & Child Reunion', 'Mother &amp; Child Reunion');
+        });
+
+        it("should escape attribute value with < character", function () {
+            eq('<output value="2 < 3" />', '<output value="2 &lt; 3" />');
+        });
+
+        it("should escape attribute value with > character", function () {
+            eq('<output value="2 > 3" />', '<output value="2 &gt; 3" />');
+        });
+
+        it("should escape attribute value with <= character", function () {
+            eq('<output value="2 <= 3" />', '<output value="2 &lt;= 3" />');
+        });
+
+        it("should escape attribute value with >= character", function () {
+            eq('<output value="2 >= 3" />', '<output value="2 &gt;= 3" />');
+        });
+
+        it("should escape attribute value with < tight", function () {
+            eq('<output value="2<3" />', '<output value="2&lt;3" />');
+        });
+
+        it("should escape attribute value with > tight", function () {
+            eq('<output value="2>3" />', '<output value="2&gt;3" />');
+        });
+
+        it("should escape < character", function () {
+            eq('your visit count must be < 5', 'your visit count must be &lt; 5');
+        });
+
+        it("should escape > character", function () {
+            eq('your visit count must be > 5', 'your visit count must be &gt; 5');
+        });
+
+        it("should escape < character before tag", function () {
+            // immortalizing https://github.com/dimagi/Vellum/pull/212
+            eq('your visit count must be < <output value="/path" />',
+               'your visit count must be &lt; <output value="/path" />');
```

Comment:
Not sure I understand how this one figures out not to convert the second < but I like the fact that we're testing that